File: Golang/MergeIntervals/MergeIntervals_test.go
package MergeIntervals

import (
	"testing"
	"reflect"
)

func TestMerge(t *testing.T) {

	if reflect.DeepEqual(merge([]Interval{{
		Start: 1,
		End:   1,
	}}), []Interval{{
		Start: 1,
		End:   1,
	}}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(merge([]Interval{{
		Start: 1,
		End:   2,
	}, {
		Start: 2,
		End:   2,
	}}), []Interval{{
		Start: 1,
		End:   2,
	}}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(merge([]Interval{{
		Start: 1,
		End:   2,
	}, {
		Start: 4,
		End:   5,
	}, {
		Start: 7,
		End:   8,
	}}), []Interval{{
		Start: 1,
		End:   2,
	}, {
		Start: 4,
		End:   5,
	}, {
		Start: 7,
		End:   8,
	}}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(merge([]Interval{{
		Start: 1,
		End:   3,
	}, {
		Start: 2,
		End:   4,
	}}), []Interval{{
		Start: 1,
		End:   4,
	}}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(merge([]Interval{{
		Start: 1,
		End:   4,
	}, {
		Start: 2,
		End:   3,
	}}), []Interval{{
		Start: 1,
		End:   4,
	}}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/MergeIntervalsTest.py
from unittest import TestCase

from MergeIntervals import MergeIntervals, Interval


class TestMergeIntervals(TestCase):
    def test_merge(self):
        mi = MergeIntervals()

        self.assertEqual(mi.merge([Interval(1, 1)]), [Interval(1, 1)])

        self.assertEqual(mi.merge([Interval(1, 2), Interval(2, 2)]), [Interval(1, 2)])

        self.assertEqual(mi.merge([Interval(1, 2), Interval(4, 5), Interval(7, 8)]),
                         [Interval(1, 2), Interval(4, 5), Interval(7, 8)])

        self.assertEqual(mi.merge([Interval(1, 3), Interval(2, 4)]),
                         [Interval(1, 4)])

        self.assertEqual(mi.merge([Interval(1, 4), Interval(2, 3)]),
                         [Interval(1, 4)])
